modules/math/shared/src/main/scala/gsp/math/Offset.scala

```diff
@@ -49,122 +51,87 @@ object Offset extends OffsetOptics {
     Order.by(o => (o.p, o.q))
 
   /** P component of an angular offset.. */
-  final case class P(toAngle: Angle) {
+  final case class Component[A](toAngle: Angle) {
 
     /** This P component, reflected around the 0 .. 180° axis. Exact, invertable. */
-    def unary_- : P =
-      P(-toAngle)
+    def unary_- : Component[A] =
+      Component[A](-toAngle)
 
     /** Some of this P component and `p`. Exact. */
-    def +(p: P): P =
-      P(toAngle + p.toAngle)
+    def +(p: Component[A]): Component[A] =
+      Component[A](toAngle + p.toAngle)
 
     /** This P component in signed radians. */
     def toRadians: Double =
       toAngle.toSignedDoubleRadians
-
   }
-  object P extends POptics {
-
-    /** The zero P component. */
-    val Zero: P =
-      P(Angle.Angle0)
-
-    /** P forms a commutative group. */
-    implicit val CommutativeGroupP: CommutativeGroup[P] =
-      new CommutativeGroup[P] {
-        val empty: P = Zero
-        def combine(a: P, b: P) = a + b
-        def inverse(a: P) = -a
+
+  object Component extends ComponentOptics {
+    /** The zero [A] component. */
+    def Zero[A]: Component[A] =
+      Component[A](Angle.Angle0)
+
+    /** Component[A] forms a commutative group. */
+    implicit def CommutativeGroupComponent[A]: CommutativeGroup[Component[A]] =
+      new CommutativeGroup[Component[A]] {
+        val empty: Component[A] = Zero[A]
+        def combine(a: Component[A], b: Component[A]) = a + b
+        def inverse(a: Component[A]) = -a
       }
 
-    implicit val ShowP: Show[P] =
+    implicit def ShowComponent[A]: Show[Component[A]] =
```

Comment:
@tpolecat Are we moving in the direction of not doing `Show`?

Reply:
My current thinking is that it's ok to define `Show` but it should always be the same thing as `toString`, which should always be for debugging and never shown to the user. Changing a `Show` or `toString` should never affect visible behavior. Use specific methods like `formatUTC` or `formatHmsDms` if you need to get a string with specific expected behavior; and define a `Format[String, A]` if there is an accompanying parser for that format. All of this is to prevent the situation we have in OCS where you never know if it's ok to change a toString, and you never know whether parsers and toStrings are consistent.